docs/general/builtins/window-functions.rst

```diff
@@ -607,6 +607,56 @@ Example::
    +---------+------+--------+-------------+
    SELECT 5 rows in set (... sec)
 
+
+.. _window-function-rank:
+
+``rank()``
+----------
+
+.. note::
+
+    The ``rank`` window function is an :ref:`enterprise feature
+    <enterprise-features>`.
+
+Synopsis
+........
+
+::
+
+    rank()
+
+Returns the rank of every row within a partition of a result set.
+
+Within each partition, the rank of the first row is ``1``. Subsequent tied
+rows are given the same rank, and the potential rank of the next row
+is incremented. Because of this, ranks may not be sequential.
+
+Example::
+
+    cr> SELECT
+    ...   col1,
+    ...   col2,
+    ...   RANK() OVER (ORDER BY col1) as rank
+    ... FROM (VALUES
```

Comment:
Do you think it would help users if we used some more realistic data set to also illustrate a common use case for the rank function?

For example, in https://www.sqltutorial.org/sql-window-functions/sql-rank/ they rank the salary.

Reply:
Yeah, that's a good idea! Will do so.